src/lib/Libattr/attr_get_set.c

```diff
@@ -0,0 +1,354 @@
+		return;
+	}
+
+	switch (op) {
+		case SET:
+			pattr->at_val.at_long = value;
+			break;
+		case INCR:
+			pattr->at_val.at_long += value;
+			break;
+		case DECR:
+			pattr->at_val.at_long -= value;
+			break;
+		default:
+			return;
+	}
+
+	pattr->at_flags |= ATR_SET_MOD_MCACHE;
+}
+
+/**
+ * @brief	Attribute setter function for char type values
+ *
+ * @param[in]	pattr	-	pointer to attribute being set
+ * @param[in]	value	-	value to be set
+ *
+ * @return	void
+ *
+ * @par MT-Safe: No
+ * @par Side Effects: None
+ *
+ */
+void
+set_attr_c(attribute *pattr, char value, enum batch_op op)
+{
+	if (pattr == NULL) {
+		log_err(-1, __func__, "Invalid pointer to attribute");
+		return;
+	}
+
+	switch (op) {
+		case SET:
+			pattr->at_val.at_char = value;
+			break;
+		case INCR:
+			pattr->at_val.at_char += value;
+			break;
+		case DECR:
+			pattr->at_val.at_char -= value;
+			break;
+		default:
+			return;
+	}
+
+	pattr->at_flags |= ATR_SET_MOD_MCACHE;
+}
+
+/**
+ * @brief	Attribute setter function for boolean type values
+ *
+ * @param[in]	pattr	-	pointer to attribute being set
+ * @param[in]	value	-	value to be set
+ *
+ * @return	void
+ *
+ * @par MT-Safe: No
+ * @par Side Effects: None
+ *
+ */
+void
+set_attr_b(attribute *pattr, long val, enum batch_op op)
+{
+	switch (op) {
+		case SET:
+			pattr->at_val.at_long = val;
+			break;
+
+		case INCR:
+			pattr->at_val.at_long = pattr->at_val.at_long | val; /* "or" */
+			break;
+
+		case DECR:
+			pattr->at_val.at_long = pattr->at_val.at_long & ~val;
+			break;
+
+		default:
+			return;
+	}
+	pattr->at_flags |= ATR_SET_MOD_MCACHE;
+}
+
+/**
+ * @brief	Mark an attribute as "not set"
+ *
+ * @param[in]	attr	-	pointer to attribute being modified
+ *
+ * @return	void
+ *
+ * @par MT-Safe: No
+ * @par Side Effects: None
+ *
+ */
+void
+mark_attr_not_set(attribute *attr)
+{
+	if (attr != NULL)
+		attr->at_flags &= ~ATR_VFLAG_SET;
+}
+
+/**
+ * @brief	Mark an attribute as "set"
+ *
+ * @param[in]	attr	-	pointer to attribute being modified
+ *
+ * @return	void
+ *
+ * @par MT-Safe: No
+ * @par Side Effects: None
+ *
+ */
+void
+mark_attr_set(attribute *attr)
+{
+	if (attr != NULL)
+		attr->at_flags |= ATR_VFLAG_SET;
+}
+
+/**
+ * @brief	Attribute getter function for char type values
+ *
+ * @param[in]	pattr	-	pointer to attribute being set
+ *
+ * @return	char
+ * @retval	char value of the attribute
+ * @retval	-1 if attribute is NULL
+ *
+ * @par MT-Safe: No
+ * @par Side Effects: None
+ */
+char
+get_attr_c(const attribute *pattr)
+{
+	if (pattr != NULL)
+		return  pattr->at_val.at_char;
+	return -1;
+}
+
+/**
+ * @brief	Attribute getter function for long type values
+ *
+ * @param[in]	pattr	-	pointer to attribute being set
+ *
+ * @return	long
+ * @retval	long value of the attribute
+ * @retval	-1 if attribute is NULL
+ *
+ * @par MT-Safe: No
+ * @par Side Effects: None
+ */
+long
+get_attr_l(const attribute *pattr)
+{
+	if (pattr != NULL)
+		return  pattr->at_val.at_long;
+	return -1;
```

Comment:
thanks for catching that .. but modifying this function to take a return pointer is going to be a very painful change to make, all the calls will have to pass it a return pointer, it's just called in way too many places and will have to be done manually. Instead of this, I'll remove the NULL check, in existing code this will not cause any issues as we were directly using pattr->at_val.at_long everywhere, so the NULL checks for pattr should be there in the right places anyways.

Reply:
Once we move to CPP, the NULL check won't be needed anyways